### PR TITLE
gcp - add cloud deployment manager initial setup

### PR DESCRIPTION
**Pull request by PMitrafanau**
Added initial class with query and get functionality for [Deployments](https://cloud.google.com/deployment-manager/docs/reference/latest/deployments) resource

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/entry.py [hunk 1]
```diff
@@ -36,8 +36,8 @@
 import c7n_gcp.resources.source
 import c7n_gcp.resources.spanner
 import c7n_gcp.resources.storage
-import c7n_gcp.resources.sql  # noqa: F401
-
+import c7n_gcp.resources.sql
+import c7n_gcp.resources.deploymentmanager  # noqa: F401
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/entry.py]: this is a sorted list, please insert the new resource in the correct position

**Comment by kapilt** [comment]
what's a custodian use case for deployments?

**Comment by PMitrafanau** [comment]
A deployment is a set of resources. If you [delete](https://console.cloud.google.com/dm/deployments/details/lamp-1?project=mitrop-custodian) the deployment, all the resources it consists of are also deleted. So I can imagine all those use cases with deletion that can be applied, for example, to compute engine: delete after some period of time, delete those having some exact label etc.

File: tools/c7n_gcp/c7n_gcp/entry.py (hunk 1 above)
Comment (PMitrafanau) [reply tools/c7n_gcp/c7n_gcp/entry.py]: done

**Pull request merged**